src/third_party/dartino/object_memory.cc

```diff
@@ -27,6 +27,10 @@ Chunk::Chunk(Space* owner, uword start, uword size, bool external)
         scavenge_pointer_(start_) {
   if (GcMetadata::in_metadata_range(start)) {
     GcMetadata::initialize_overflow_bits_for_chunk(this);
```

Comment:
It feels like you should share the code for the initialization of chunks. You have two different versions of this (one with 2 initialization calls and one with 3). Maybe add a shared initializer function that takes a parameter instead?